wdl/GenotypeDepthPart2.wdl

```diff
@@ -91,50 +91,52 @@ workflow GenotypeDepthPart2 {
     }
   }
 
-  scatter (lt5kb_bed in SplitVariants.lt5kb_beds) {
-
-    call tasksgenotypebatch.RDTestGenotype as RDTestGenotypeUnder5kb {
-      input:
-        bin_exclude=bin_exclude,
-        bin_exclude_idx=bin_exclude_idx,
-        bed = lt5kb_bed,
-        coveragefile = coveragefile,
-        coveragefile_index = coveragefile_index,
-        medianfile = medianfile,
-        samples = samples,
-        gt_cutoffs = RD_pesr_sepcutoff,
-        n_bins = n_RdTest_bins,
-        prefix = basename(lt5kb_bed, ".bed"),
-        generate_melted_genotypes = true,
-        ref_dict = ref_dict,
-        sv_pipeline_rdtest_docker = sv_pipeline_rdtest_docker,
-        runtime_attr_override = runtime_attr_rdtest_genotype
-    }
-
-    call tasksgenotypebatch.MakeSubsetVcf as MakeSubsetVcfUnder5kb {
-      input:
-        vcf = cohort_vcf,
-        bed = lt5kb_bed,
-        sv_base_mini_docker = sv_base_mini_docker,
-        runtime_attr_override = runtime_attr_make_subset_vcf
-    }
-
-    call tasksgenotypebatch.IntegrateDepthGq as IntegrateDepthGqUnder5kb {
-      input:
-        vcf = MakeSubsetVcfUnder5kb.subset_vcf,
-        RD_melted_genotypes = RDTestGenotypeUnder5kb.melted_genotypes,
-        RD_vargq = RDTestGenotypeUnder5kb.varGQ,
-        sv_pipeline_docker = sv_pipeline_docker,
-        runtime_attr_override = runtime_attr_integrate_depth_gq
-    }
-    call tasksgenotypebatch.AddGenotypes as AddGenotypesUnder5kb {
-      input:
-        vcf = MakeSubsetVcfUnder5kb.subset_vcf,
-        genotypes = IntegrateDepthGqUnder5kb.genotypes,
-        varGQ = IntegrateDepthGqUnder5kb.varGQ,
-        prefix = basename(lt5kb_bed, ".bed"),
-        sv_pipeline_docker = sv_pipeline_docker,
-        runtime_attr_override = runtime_attr_add_genotypes
+  if (length(SplitVariants.lt5kb_beds) > 0) {
```

Comment:
Is not `select_all` you added below sufficient? I think it would be useful to test without this `if`, because if this turns out that this `if` is a requirement, it could happen on other workflows where we have shards like this.

Reply:
I haven't tested it without the `if` so I can't say for sure. This was my logic: 

- Without the `if`, the output arrays from the scatters are both required/defined - the lt5kb one simply may be empty. `select_all` selects all of the items in the array that are defined - but they were already both defined, so that didn't seem like it would be sufficient.
- It wasn't clear whether the cromwell bug was related to trying to initiate an empty scatter, or if it occurred during `flatten` when attempting to gather the empty scatter. The `if` covers both cases.

It is possible that there are other cases like this. We have observed this cromwell issue most often in GenotypeDepthPart2, but I know it has occurred in at least one other workflow before.

I agree that testing with and without the `if` could provide more insight into this issue. I don't have bandwidth to do that right now so I just wanted to get something that works.

Reply:
The `if` statement seems to be needed due to a Cromwell issue, and that necessitates the `select_all` below. `select_all` only acts on optional types and would have no effect if the `if` weren't present.

Reply:
This is a ubiquitous pattern and it seems we're been very lucky so far that other workflows run with no issues.

Reply:
It's not luck - it's rare that an array over which we scatter is empty like this